src/main/java/rodain/source/ui/FileExplorerPane.java

```diff
@@ -0,0 +1,402 @@
+package rodain.source.ui;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.*;
+
+import javafx.application.Platform;
+import javafx.event.ActionEvent;
+import javafx.event.EventHandler;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.control.*;
+import javafx.scene.input.*;
+import javafx.scene.layout.*;
+import javafx.stage.DirectoryChooser;
+import javafx.stage.Stage;
+import javafx.util.Callback;
+
+import org.slf4j.LoggerFactory;
+
+import rodain.core.Footer;
+import rodain.source.representation.SourceDirectory;
+import rodain.source.ui.items.SourceTreeDirectory;
+import rodain.source.ui.items.SourceTreeItem;
+import rodain.source.ui.items.SourceTreeItemState;
+import rodain.utils.Utils;
+import rodain.utils.WalkFileTree;
+
+/**
+ * Created by adrapereira on 24-09-2015.
+ */
+public class FileExplorerPane extends BorderPane implements Observer {
+    private static final org.slf4j.Logger log = LoggerFactory.getLogger(FileExplorerPane.class.getName());
+    private HBox top;
+    private StackPane fileExplorer;
+    private TreeView<String> treeView;
+    private Stage stage;
+
+    private HBox filterButtons;
+    private ComputeDirectorySize computeSize;
+
+    private static Set<String> oldIgnored;
+    private static Set<String> oldMapped;
+
+    //Filter control
+    private static boolean showFiles = true;
+    private static boolean showIgnored = false;
+    private static boolean showMapped = false;
+
+    //This thread is used to walk a directory's file tree and update the UI periodically with the SIZE and file count
+    private WalkFileTree computeThread;
+
+    public FileExplorerPane(Stage stage){
+        super();
+
+        this.stage = stage;
+
+        createTop();
+        createFileExplorer();
+        createFilterButtons();
+
+        setFileExplorerRoot(Paths.get("/home/adrap/Documents/Git/roda-in/"));
```

Comment:
Should never, ever, put hardcoded paths in code.
Use System.getProperty("user.home") instead.